### PR TITLE
MetricReporters now use MetricReporterFilter to specify what stats to re...

### LINKED ISSUE #175: Do away with querying in MetricReporters
Now that we don't need metric filters to squash or combine metrics in Reporters, we should switch MetricReporters to indicating what metrics to report in either a white list, black list.

### PR DESCRIPTION
**Pull request by nsauro**
...port, no longer uses a Seq[MetricFilter].  Also, removed unused classes.

fixes #175

### CODE REVIEW

File: colossus-metrics/src/main/scala/colossus/metrics/StatReporter.scala [hunk 1]
```diff
@@ -64,6 +62,19 @@ class MetricReporter(intervalAggregator : ActorRef, config: MetricReporterConfig
     }
   }
 
+  private def filterMetrics(m : MetricMap) : RawMetricMap = {
+    filters match {
+      case MetricReporterFilter.All => m.toRawMetrics
+      case MetricReporterFilter.None => Map()
```
Comment (DanSimon) [inline colossus-metrics/src/main/scala/colossus/metrics/StatReporter.scala]: Will anyone actually ever use None?  Seems kinda weird if someone made a reporter and didn't report anything.

Comment (nsauro) [inline colossus-metrics/src/main/scala/colossus/metrics/StatReporter.scala]: Yea..i I figured as much..seems totally weird.  I just put it in in the even that they just wanted to quickly silence metric reporting.  I guess they could just comment out that line.  I can remove that case.

**Comment by DanSimon** [comment]
Aside from my one point, this looks good

**Comment by nsauro** [comment]
Removed the None option.,  Will merge this once travis greenlights it

**Pull request merged**